x/evm/keeper/fees.go

```diff
@@ -55,8 +55,28 @@ func (k *Keeper) DeductTxCostsFromUserBalance(
 		return errorsmod.Wrapf(err, "account not found for sender %s", from)
 	}
 
+	var (
+		params   = k.GetParams(ctx)
+		evmDenom = params.EvmDenom
+		denomDec = params.DenomDecimals
+	)
+
+	if found, _ := fees.Find(evmDenom); !found {
+		return errorsmod.Wrapf(errortypes.ErrInsufficientFee, "fee token %s not found in the fees", evmDenom)
+	}
+
+	// only evmDenom is allowed on fee coins
+	if fees.Len() > 1 {
+		return errorsmod.Wrapf(errortypes.ErrInvalidCoins, "fee token can only be %s", evmDenom)
+	}
```

Comment:
nit: we should change the order of these two checks to avoid iterating all coins if `len() > 1`